source/server/server.cc

```diff
@@ -67,7 +67,7 @@ InstanceImpl::InstanceImpl(
       handler_(new ConnectionHandlerImpl(*dispatcher_, "main_thread")),
       random_generator_(std::move(random_generator)), listener_component_factory_(*this),
       worker_factory_(thread_local_, *api_, hooks),
-      dns_resolver_(dispatcher_->createDnsResolver({})),
+      dns_resolver_(dispatcher_->createDnsResolver({}, false)),
```

Comment:
What is this DNS resolver used for? Do we need to support this option for these use cases?

Reply:
@htuch good catch. After some code investigation, turns our this is a default resolver (/etc/resolv.conf) that is used by specific type of clusters if `dns_resolvers` field is not configured. In my opinion, newly introduced config option should affect user specified resolvers (configured via `dns_resolvers`) and this default resolver.

Reply:
Presumably this is a server global config option, i.e. in bootstrap?

Reply:
Suggestion: add `use_tcp_for_dns_lookups` to bootstrap config (so it will be used in server's dns resolver) and make it possible to override `use_tcp_for_dns_lookups` per cluster in cds config.

Reply:
Sounds like a plan.